MongoUsingMongoose/controller.js

```diff
@@ -0,0 +1,205 @@
+exports.update = (req, res) => {
+  // Validate Request
+  if(!req.body.email) {
+      return res.status(400).send({
+          message: "User email can not be empty"
+      });
+  }
+
+  // Find user and update it with the request body
+  User.findOneAndUpdate({email :req.params.email}, {
+      id : req.body.id || "Untitled user",
+      name : req.body.name,
+      email : req.body.email,
+      location :req.body.location
+  }, {new: true})
+  .then(user => {
+      if(!user) {
+          return res.status(404).send({
+              message: "User not found with email " + req.params.email
+          });
+      }
+      res.send(user);
+  }).catch(err => {
+      if(err.kind === 'ObjectId') {
+          return res.status(404).send({
+              message: "User not found with email " + req.params.email
+          });                
+      }
+      return res.status(500).send({
+          message: "Error updating user with email " + req.params.email
+      });
+  });
+};
+
+// Delete a user with the specified email in the request
+exports.delete = (req, res) => {
+  User.findOneAndDelete(req.params.email)
+  .then(user => {
+      if(!user) {
+          return res.status(404).send({
+              message: "User not found with email" + req.params.email
+          });
+      }
+      res.send({message: "User deleted successfully!"});
+  }).catch(err => {
+      if(err.kind === 'ObjectId' || err.name === 'NotFound') {
+          return res.status(404).send({
+              message: "User not found with email " + req.params.email
+          });                
+      }
+      return res.status(500).send({
+          message: "Could not delete user with email " + req.params.email
+      });
+  });
+};
+
+exports.updatePassword = (req,res) =>
+{
+     // Find user and update it with the request body
+  User.findOneAndUpdate({email : req.params.email}, {
+    password : req.body.password
+    }, {new: true})
+    .then(user => {
+    if(!user) {
+        return res.status(404).send({
+            message: "User not found with email " + req.params.email
+        });
+    }
+    res.send(user);
+    }).catch(err => {
+    if(err.kind === 'ObjectId') {
+        return res.status(404).send({
+            message: "User not found with email " + req.params.email
+        });                
+    }
+    return res.status(500).send({
+        message: "Error updating user with email " + req.params.email
+     });
+    });
+
+}
+
+exports.checkUser = (req,res) =>{
+  User.findOne({email :req.body.email})
+  .then(user => {
+      if(!user) {
+          return res.status(404).send({
+              message: "User not found with email " + req.params.email
+          });            
+      }  
+      if((req.body.email == user.email) && (req.body.password == user.password))
+      {
+          const username = {name : user.name};
+          const accessToken = jwt.sign(username,process.env.ACCESSTOKEN,{expiresIn : '40s'});
```

Comment:
Token should be generated and sent back to user. Not from ENV

Reply:
changed the name of the variable that was creating confusion